subsys/bluetooth/controller/ll_sw/ull.c

```diff
@@ -332,10 +332,50 @@ int ll_init(struct k_sem *sem_rx)
 	return  0;
 }
 
+static void perform_lll_reset(void *param)
+{
+	int err;
+
+	/* Reset LLL */
+	err = lll_reset();
+	LL_ASSERT(!err);
+
+#if defined(CONFIG_BT_BROADCASTER)
+	/* Reset adv state */
+	err = lll_adv_reset();
+	LL_ASSERT(!err);
+#endif /* CONFIG_BT_BROADCASTER */
+
+#if defined(CONFIG_BT_OBSERVER)
+	/* Reset scan state */
+	err = lll_scan_reset();
+	LL_ASSERT(!err);
+#endif /* CONFIG_BT_OBSERVER */
+
+#if defined(CONFIG_BT_CONN)
+	/* Reset conn role */
+	err = lll_conn_reset();
+	LL_ASSERT(!err);
+#endif /* CONFIG_BT_CONN */
+}
+
 void ll_reset(void)
 {
 	int err;
 
+	/* Reset LLL via mayfly */
```

Comment:
Move to around line 415, after call to ```ull_conn_reset```. Let the ull resets get the states/roles down gracefully before LLL resets are called.